pkg/workloads/console/controller.go

```diff
@@ -78,6 +82,54 @@ func (r *ConsoleReconciler) Reconcile(request reconcile.Request) (res reconcile.
 		return res, err
 	}
 
-	logger.Log("event", "Reconciled") // TODO
-	return
+	logger.Log("msg", "found console")
+
+	// Fetch the template for this console
+	consoleTemplateName := types.NamespacedName{
+		Name:      csl.Spec.ConsoleTemplateRef.Name,
+		Namespace: request.NamespacedName.Namespace,
+	}
+	consoleTemplate := &workloadsv1alpha1.ConsoleTemplate{}
+	if err := r.client.Get(r.ctx, consoleTemplateName, consoleTemplate); err != nil {
+		return res, err
+	}
+
+	// Try to find the job for this console
+	logger.Log("msg", "looking for job")
```

Comment:
Can we simplify this flow by using `controllerutil.CreateOrUpdate` and just doing that to our job?

Reply:
For reference: https://github.com/gocardless/theatre/blob/fbf6d90da93c85395d06f5f05d8a33ccc784d86b/pkg/controllers/sudorolebinding/controller.go#L245-L270

Reply:
As discussed - we'll keep on eye on this pattern as we build out the operator. We may need it a lot more for rolebindings.